charts/velero/templates/prometheusrule.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.metrics.enabled .Values.metrics.prometheusRule.enabled }}
+{{- if and (and .Values.metrics.enabled .Values.metrics.prometheusRule.enabled) (or (not .Values.metrics.prometheusRule.autodetect) (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1")) (.Values.metrics.prometheusRule.spec) }}
```

Comment:
Is the (.Values.metrics.prometheusRule.spec) purpose to check the .Values.metrics.prometheusRule.spec exists, not nil?

Reply:
Yes, exactly. If it's nil / an empty dict the rule would have been created but without any actual rules inside. This way it will only create if enabled, capabilities are met and some rule specs are defined.